code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -118,6 +118,136 @@ public static void updatePropositions(
+        eventData.put(
+                OptimizeConstants.EventDataKeys.REQUEST_TYPE,
+                OptimizeConstants.EventDataValues.REQUEST_TYPE_UPDATE);
+        eventData.put(OptimizeConstants.EventDataKeys.DECISION_SCOPES, flattenedDecisionScopes);
+
+        if (!OptimizeUtils.isNullOrEmpty(xdm)) {
+            eventData.put(OptimizeConstants.EventDataKeys.XDM, xdm);
+        }
+
+        if (!OptimizeUtils.isNullOrEmpty(data)) {
+            eventData.put(OptimizeConstants.EventDataKeys.DATA, data);
+        }
+
+        final Event event =
+                new Event.Builder(
+                                OptimizeConstants.EventNames.UPDATE_PROPOSITIONS_REQUEST,
+                                OptimizeConstants.EventType.OPTIMIZE,
+                                OptimizeConstants.EventSource.REQUEST_CONTENT)
+                        .setEventData(eventData)
+                        .build();
+
+        MobileCore.dispatchEventWithResponseCallback(
+                event,
+                OptimizeConstants.GET_RESPONSE_CALLBACK_TIMEOUT,
+                new AdobeCallbackWithError<Event>() {
+                    @Override
+                    public void fail(final AdobeError adobeError) {
+                        AEPOptimizeError aepOptimizeError =
+                                new AEPOptimizeError(null, null, null, null, adobeError);
+                        failWithOptimizeError(callback, aepOptimizeError);
+                    }
+
+                    @Override
+                    public void call(final Event event) {
+                        try {
+                            final Map<String, Object> eventData = event.getEventData();
+                            if (OptimizeUtils.isNullOrEmpty(eventData)) {
+                                AEPOptimizeError aepOptimizeError =
+                                        new AEPOptimizeError(
+                                                null,
+                                                null,
+                                                null,
+                                                null,
+                                                AdobeError.UNEXPECTED_ERROR);
+                                failWithOptimizeError(callback, aepOptimizeError);
+                                return;
+                            }
+
+                            if (eventData.containsKey(
+                                    OptimizeConstants.EventDataKeys.RESPONSE_ERROR)) {
+                                final int errorCode =
+                                        DataReader.getInt(
+                                                eventData,
+                                                OptimizeConstants.EventDataKeys.RESPONSE_ERROR);
+                                failWithOptimizeError(
+                                        callback,
+                                        new AEPOptimizeError(
+                                                null,
```

Comment:
Why are these values always null? Is this PR incomplete? We're supposed to send a timeout error here from `OptimizeExtension.java` right?

Reply:
updated these fields